src/Extensions/ServiceFabricGuest.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Omex.Extensions.ServiceFabricGuest.Abstractions
+{
+	/// <summary>
+	/// Extension to add Service Fabric Client dependencies to IServiceCollection
+	/// </summary>
+	public static class ServiceCollectionExtensions
+	{
+		/// <summary>
+		/// Register types required Service Fabric comunication
+		/// </summary>
+		public static IServiceCollection AddServiceFabricClient(this IServiceCollection serviceCollection)
+		{
+			return serviceCollection.AddServiceFabricClient<ServiceFabricClientWrapper>();
+		}
+
+		/// <summary>
+		/// Register generic types required Service Fabric comunication
+		/// </summary>
+		public static IServiceCollection AddServiceFabricClient<TClinet>(this IServiceCollection serviceCollection)
+			where TClinet : class, IServiceFabricClientWrapper
+		{
+			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IServiceFabricClientWrapper, TClinet>());
```

Comment:
`TryAddEnumerable` used only when we might use multiple implementations at the same time, like sending health into multiple publishers. Here just using `serviceCollection.TryAddSingleton<IServiceFabricClientWrapper, TClinet>();` should be better.